base/file.jl

```diff
@@ -682,7 +709,10 @@ function mktempdir(fn::Function, parent::AbstractString=tempdir();
         fn(tmpdir)
     finally
         try
-            ispath(tmpdir) && rm(tmpdir, recursive=true)
+            if ispath(tmpdir)
+                prepare_for_deletion(tmpdir)
+                rm(tmpdir, recursive=true)
```

Comment:
Is there a reason we weren't passing `force=true` here, which already includes some logic to deal with this failure case? (and possibly extend that logic to cover more cases)

Reply:
I don't know why we weren't passing `force=true`.